python/hsfs/engine/python.py

```diff
@@ -364,13 +371,20 @@ def _read_hopsfs_remote(self, location, data_format, read_options=None):
                     ):
                         arrow_flight_config = read_options.get("arrow_flight_config")
                         df = arrow_flight_client.get_instance().read_path(
-                            inode.path, arrow_flight_config
+                            inode.path,
+                            arrow_flight_config,
+                            dataframe_type=dataframe_type,
```

Comment:
Why do you need to pass dataframe type here and use it again below?

Reply:
Arrow flight returns the dataframe as pandas by default. I had modifed `_get_dataset` function in _arrow_flight_client.py_ to read as polars dataframe when the dataframe type is polars. So this passing of `dataframe_type` is used to acheive this functionality.